### ISSUE TITLE
perlin_noise is a pain to work with

### ISSUE THREAD

**Issue by jgostick** [PMEAL/porespy#372]
The perlin_noise function has the potential to be the next blobs, but it has a problem with accepting different size images.  This should be fixed.  Perhaps the image should be increased to the next possible size, then cropped back down?

**Comment by ma-sadeghi** [comment]
It also doesn't respect numpy's random seed.

**Comment by jgostick** [comment]
Just discovered a SUPERB package for generating fractal noise:  https://github.com/robbmcleod/pyfastnoisesimd

The usage is object oriented, as shown below.  Wrapping in the ``perlin_noise`` function would make sense, but we should make users aware of the actual package as well since it seems to be very powerful.

```python

import pyfastnoisesimd as fns
import numpy as np
import porespy as ps
import matplotlib.pyplot as plt

shape = [500,512,512]
seed = np.random.randint(2**31)
N_threads = 16

perlin = fns.Noise(seed=seed, numWorkers=N_threads)
perlin.frequency = 0.02
perlin.noiseType = fns.NoiseType.SimplexFractal
perlin.fractal.octaves = 6
perlin.fractal.lacunarity = 2
perlin.fractal.gain = .5
# perlin.perturb.perturbType = fns.PerturbType.NoPertrub
result = perlin.genAsGrid(shape)
# result = ps.tools.norm_to_uniform(result, scale=[0, 1])
ps.imshow(result)

```

**Comment by jgostick** [comment]
fixed in #423
